### PR TITLE
fix: enable proper initialization of c++ global objects

### PR DESCRIPTION
**Pull request by bfredl**
not done yet, this is just a demo that it work.

most of the details here were provided by @gpanders , I just adapted them to the deluge code base.

### CODE REVIEW

**Comment by bfredl** [comment]
some references:

 -  https://interrupt.memfault.com/blog/how-to-write-linker-scripts-for-firmware
 -  https://ftp.gnu.org/old-gnu/Manuals/ld-2.9.1/html_chapter/ld_3.html

**Comment by bfredl** [comment]
Note: there are more global constructors in `SoundEditor::SoundEditor()` but I haven't touched these as then the logic to assign names to menu items would need to be changed. This should be done as a separate task (likely involving other refactors of the menu system), and is out scope for this PR. For now, letting these objects be re-initialized is perfectly fine (there is no memory leak, as the default constructors do not allocate memory).

**Comment by jamiefaye** [comment]
I checked out this PR and built it for my 7 Seg Deluge and encountered a problem when I scrolled to a synth and pushed the select button. This caused the system to freeze. Hanging up at MenuItemSubmenu::drawPixelsForOled() at the call to drawName().

**Comment by bfredl** [comment]
> I checked out this PR and built it for my 7 Seg Deluge and encountered a problem when I scrolled to a synth and pushed the select button. This caused the system to freeze. Hanging up at MenuItemSubmenu::updateDisplay() at the call to drawName().

Could reproduce some crashes on the OLED as well. Pushed a workaround for those.

**Comment by jamiefaye** [comment]
Issue is still present on 7SEG.

File: linker_script_rz_a1l.ld [hunk 1]
```diff
@@ -186,32 +186,27 @@ SECTIONS
...
-		_mdata = .;
-		*/
-		
 		*(.got.plt) 
 		*(.got)
 		.	= ALIGN(0x8);
 		address_end_data_ROM = .;
 	} > RAM012L
-	
+
```
Comment (stellar-aria) [inline linker_script_rz_a1l.ld:194]: I think you'll also need a `.preinit_array` section in here to deal with vtable stuff (validation and maybe init?)

References: 
[Picolibc docs (but also applies for newlib)](https://github.com/picolibc/picolibc/blob/main/doc/init.md)
[Info on the SysV ABI](https://stackoverflow.com/a/60178049)

Comment (bfredl) [reply linker_script_rz_a1l.ld:194]: thanks, done

**Comment by bfredl** [comment]
superseeded by #104

**Pull request closed without merging**